### DISCUSSION TITLE
Get access to texture's image after allocation

### DISCUSSION THREAD

**Discussion by EduMenges** [emilk/egui#2802] [Q&A] [answered]
I'm working on a collage project for image processing, but I'm struggling with the textures.

After loading, I need to do the actual processing. My plan is to just access the image data, copy it, and use it with the [https://crates.io/crates/image](image crate), but after a lot of searching, I really don't know how. I think I'll just keep two copies, one for the texture, and another for the image, tho that would not be memory efficient.

There could also be a way to consume a texture and return its underlying ImageData, so no copy would be necessary. But I really, really don't know how.

**Comment by John-Nagle** [comment, marked as answer]
Usually, you load your own image and then put it into egui. In some uses of egui, that means it gets sent to the GPU, and it's no longer in main memory. You may be able to get it back from the GPU or do image processing on it there, or keep your own copy, but none of that is something egui does.

**Reply by EduMenges** [reply]
Thanks for the answer! I this case, I'll just keep a copy. It's just some megabytes more, anyway.
